File: pagertabsindicator/src/main/java/com/hold1/pagertabsindicator/TabView.java
package com.hold1.pagertabsindicator;

import android.animation.ValueAnimator;
import android.content.Context;
import android.graphics.Color;
import android.util.Log;
import android.view.MotionEvent;
import android.widget.FrameLayout;

/**
 * Created by Cristian Holdunu on 09/11/2017.
 */

public class TabView extends FrameLayout {
    private static final String TAG = TabView.class.getSimpleName();

    private ValueAnimator bgAnimator;
    private int pressColor = Color.TRANSPARENT;
    private int currentBgColor = pressColor;

    public TabView(Context context) {
        super(context);
        bgAnimator = ValueAnimator.ofInt(100, 0);
        bgAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
            @Override
            public void onAnimationUpdate(ValueAnimator animation) {
                setBackgroundColor(Color.argb((Integer) animation.getAnimatedValue(), Color.red(pressColor), Color.green(pressColor), Color.blue(pressColor)));
            }
        });
    }

    @Override
    public void setSelected(boolean selected) {

    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        switch (event.getAction()) {
            case MotionEvent.ACTION_DOWN:
                bgAnimator.cancel();
                setBackgroundColor(pressColor);
                break;
            case MotionEvent.ACTION_UP:
                if (bgAnimator != null)
                    bgAnimator.start();
                break;
        }
        return super.onTouchEvent(event);
    }

    public void onOffset(float offset) {
        Log.d(TAG, "onOffset=" + offset);
    }
}

File: pagertabsindicator/src/main/java/com/hold1/pagertabsindicator/TabView.kt
package com.hold1.pagertabsindicator

import android.widget.FrameLayout
import android.animation.ValueAnimator
import android.content.Context
import android.graphics.Color
import android.util.AttributeSet
import android.view.View

/**
 * Created by Cristian Holdunu on 09/11/2017.
 */
open class TabView @JvmOverloads constructor(
    context: Context, attrs: AttributeSet? = null
) : FrameLayout(context, attrs) {
    private val bgAnimator: ValueAnimator? = null
    private val pressColor = Color.TRANSPARENT
    private val currentBgColor = pressColor
    private var offset = 0f

    init {
        setWillNotDraw(false)
    }

    open fun onOffset(offset: Float) {
        this.offset = offset
        invalidate()
    }

    companion object {
        private val TAG = TabView::class.java.simpleName
    }
}